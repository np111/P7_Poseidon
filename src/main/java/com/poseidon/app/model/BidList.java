package com.poseidon.app.model;

import java.time.ZonedDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class BidList {
    private Long id;
    private String account;
    private String type;
    private Double bidQuantity;
    private Double askQuantity;
    private Double bid;
    private Double ask;
    private String benchmark;
    private ZonedDateTime bidListDate;
    private String commentary;
    private String security;
    private String status;
    private String trader;
    private String book;
    private String creationName;
    private ZonedDateTime creationDate;
    private String revisionName;
    private ZonedDateTime revisionDate;
    private String dealName;
    private String dealType;
    private String sourceListId;
    private String side;
}
