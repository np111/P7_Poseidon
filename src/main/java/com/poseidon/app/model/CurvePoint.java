package com.poseidon.app.model;

import java.time.ZonedDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class CurvePoint {
    private Long id;
    private Long curveId;
    private ZonedDateTime asOfDate;
    private Double term;
    private Double value;
    private ZonedDateTime creationDate;
}
