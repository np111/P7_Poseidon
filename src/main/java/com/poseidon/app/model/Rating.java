package com.poseidon.app.model;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class Rating {
    private Long id;

    @NotBlank(message = "MoodysRating is mandatory")
    private String moodysRating;

    @NotBlank(message = "SandPRating is mandatory")
    private String sandPRating;

    @NotBlank(message = "FitchRating is mandatory")
    private String fitchRating;

    @NotNull(message = "Order is mandatory")
    private Byte orderNumber;
}
