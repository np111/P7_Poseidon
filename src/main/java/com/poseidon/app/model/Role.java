package com.poseidon.app.model;

public enum Role {
    USER,
    ADMIN,
}
