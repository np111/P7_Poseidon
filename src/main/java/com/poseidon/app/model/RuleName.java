package com.poseidon.app.model;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class RuleName {
    private Long id;
    private String name;
    private String description;
    private String json;
    private String template;
    private String sqlStr;
    private String sqlPart;
}
