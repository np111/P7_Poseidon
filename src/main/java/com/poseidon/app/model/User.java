package com.poseidon.app.model;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class User {
    private Long id;

    @NotBlank(message = "Username is mandatory")
    private String username;

    @NotBlank(message = "Password is mandatory")
    @Size(min = 8, message = "Password must be at least 8 characters long")
    @Pattern(regexp = ".*\\p{Lu}.*", message = "Password must contain at least one uppercase letter.")
    @Pattern(regexp = ".*[0-9].*", message = "Password must contain at least one number.")
    @Pattern(regexp = ".*[^\\p{L}0-9].*", message = "Password must contain at least one symbol.")
    private String password;

    @NotBlank(message = "Full Name is mandatory")
    private String fullname;

    @NotNull(message = "Role is mandatory")
    private Role role;
}
