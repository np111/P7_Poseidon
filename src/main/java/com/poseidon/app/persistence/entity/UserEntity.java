package com.poseidon.app.persistence.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "users")
@NoArgsConstructor
@Data
public class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "username")
    private String username;

    @Column(name = "password", columnDefinition = "CHAR", length = 60)
    private String password;

    @Column(name = "fullname")
    private String fullname;

    @Column(name = "role")
    private String role;
}
