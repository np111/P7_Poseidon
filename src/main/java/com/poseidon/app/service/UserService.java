package com.poseidon.app.service;

import com.poseidon.app.model.User;
import com.poseidon.app.persistence.entity.UserEntity;
import com.poseidon.app.persistence.mapper.UserMapper;
import com.poseidon.app.persistence.repository.UserRepository;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor(onConstructor = @__(@Autowired))
@Service
@Scope("singleton")
public class UserService extends CrudService<User, UserEntity, Long> {
    private final @Getter UserRepository repository;
    private final @Getter UserMapper mapper;

    @Override
    protected void setModelId(User user, Long id) {
        user.setId(id);
    }
}
