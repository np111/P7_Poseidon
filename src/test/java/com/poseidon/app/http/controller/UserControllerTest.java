package com.poseidon.app.http.controller;

import com.poseidon.app.model.Role;
import com.poseidon.app.model.User;
import com.poseidon.app.persistence.entity.UserEntity;
import com.poseidon.app.service.UserService;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

@WebMvcTest(UserController.class)
class UserControllerTest extends AbstractCrudControllerTest<UserService, User, UserEntity, Long> {
    @Autowired
    private UserController userController;

    public UserControllerTest() {
        super("/user", User.class, Long.class);
    }

    @Override
    protected UserService getService() {
        return userService;
    }

    @Override
    protected User getModelA() {
        return User.builder()
                .id(1L)
                .fullname("fullname")
                .username("username")
                .password("passwordA1$")
                .role(Role.USER)
                .build();
    }

    @Override
    protected User getModelB() {
        return User.builder()
                .id(32402L)
                .fullname("yJnkKqFBtY")
                .username("xyuwd3sbk9")
                .password("passwordA1$")
                .role(Role.ADMIN)
                .build();
    }

    @Test
    void filterIn() {
        User user = new User();

        user.setPassword("xxx");
        userController.filterIn(user);
        assertTrue(user.getPassword().startsWith("$2b$"));
    }

    @Test
    void filterOut() {
        User user = new User();

        user.setPassword("");
        userController.filterOut(user);
        assertEquals("", user.getPassword());

        user.setPassword("xxx");
        userController.filterOut(user);
        assertEquals("", user.getPassword());

        user.setPassword(null);
        userController.filterOut(user);
        assertNull(user.getPassword());
    }
}
