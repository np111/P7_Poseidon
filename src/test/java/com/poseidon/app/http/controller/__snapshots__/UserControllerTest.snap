com.poseidon.app.http.controller.UserControllerTest.add=[
-
> GET /user/add
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Add New User</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/user/validate\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"fullname\" class=\"col-sm-2 control-label\">Full Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"fullname\" placeholder=\"Full Name\" class=\"col-4\" name=\"fullname\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"username\" class=\"col-sm-2 control-label\">User Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"username\" placeholder=\"User Name\" class=\"col-4\" name=\"username\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"password\" class=\"col-sm-2 control-label\">Password</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"password\" id=\"password\" placeholder=\"Password\" class=\"col-4\" name=\"password\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"password\" class=\"col-sm-2 control-label\">Role</label>"
"        <div class=\"col-sm-10\">"
"          <label class=\"checkbox-inline\"><input type=\"radio\" value=\"ADMIN\" id=\"role1\" name=\"role\">Admin</label>"
"          <label class=\"checkbox-inline\"><input type=\"radio\" value=\"USER\" id=\"role2\" name=\"role\">User</label>"
"          "
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/user/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Add user\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.UserControllerTest.delete=[
-
> GET /user/delete/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Not Found Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\"> test-username </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">Entity not found.</p>"
"</body>"
"</html>"
-
> GET /user/delete/32402
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /user/list
""
]


com.poseidon.app.http.controller.UserControllerTest.list=[
-
> GET /user/list
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" type=\"text/css\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
"  <div class=\"row\">"
"    <div class=\"col-6\">"
""
"    </div>"
"    <div class=\"col-6 text-right\">"
"      <a href=\"/\">Home</a>&nbsp;|&nbsp;<a href=\"/login\">Login</a>"
"    </div>"
"  </div>"
"  <div class=\"row\"><h2>User List</h2></div>"
"  <div class=\"row\">"
"    <a href=\"/user/add\" class=\"btn btn-primary btn-sm\">Add New</a>"
"    <table class=\"table table-bordered\">"
"      <thead>"
"      <tr>"
"        <th>Id</th>"
"        <th>Full Name</th>"
"        <th>User Name</th>"
"        <th>Role</th>"
"        <th>Action</th>"
"      </tr>"
"      </thead>"
"      <tbody>"
"      <tr>"
"        <td style=\"width: 10%\">1</td>"
"        <td>fullname</td>"
"        <td>username</td>"
"        <td style=\"width: 25%\">USER</td>"
"        <td style=\"width: 15%\" class=\"text-center\">"
"          <a href=\"/user/update/1\">Edit</a>&nbsp;|&nbsp;"
"          <a href=\"/user/delete/1\">Delete</a>"
"        </td>"
"      </tr>"
"      <tr>"
"        <td style=\"width: 10%\">32402</td>"
"        <td>yJnkKqFBtY</td>"
"        <td>xyuwd3sbk9</td>"
"        <td style=\"width: 25%\">ADMIN</td>"
"        <td style=\"width: 15%\" class=\"text-center\">"
"          <a href=\"/user/update/32402\">Edit</a>&nbsp;|&nbsp;"
"          <a href=\"/user/delete/32402\">Delete</a>"
"        </td>"
"      </tr>"
"      </tbody>"
"    </table>"
"  </div>"
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.UserControllerTest.showUpdate=[
-
> GET /user/update/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Not Found Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\"> test-username </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">Entity not found.</p>"
"</body>"
"</html>"
-
> GET /user/update/32402
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Update User</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/user/update/32402\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"fullname\" class=\"col-sm-2 control-label\">Full Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"fullname\" placeholder=\"Full Name\" class=\"col-4\" name=\"fullname\" value=\"yJnkKqFBtY\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"username\" class=\"col-sm-2 control-label\">User Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"username\" placeholder=\"User Name\" class=\"col-4\" name=\"username\" value=\"xyuwd3sbk9\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"password\" class=\"col-sm-2 control-label\">Password</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"password\" id=\"password\" placeholder=\"Password\" class=\"col-4\" name=\"password\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"password\" class=\"col-sm-2 control-label\">Role</label>"
"        <div class=\"col-sm-10\">"
"          <label class=\"checkbox-inline\"><input type=\"radio\" value=\"ADMIN\" id=\"role1\" name=\"role\" checked=\"checked\">Admin</label>"
"          <label class=\"checkbox-inline\"><input type=\"radio\" value=\"USER\" id=\"role2\" name=\"role\">User</label>"
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <input type=\"hidden\" id=\"id\" name=\"id\" value=\"32402\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/user/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Update user\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.UserControllerTest.update=[
-
> POST /user/update/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Update User</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/user/update/1\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"fullname\" class=\"col-sm-2 control-label\">Full Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"fullname\" placeholder=\"Full Name\" class=\"col-4\" name=\"fullname\" value=\"\">"
"          <p class=\"text-danger\">Full Name is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"username\" class=\"col-sm-2 control-label\">User Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"username\" placeholder=\"User Name\" class=\"col-4\" name=\"username\" value=\"\">"
"          <p class=\"text-danger\">Username is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"password\" class=\"col-sm-2 control-label\">Password</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"password\" id=\"password\" placeholder=\"Password\" class=\"col-4\" name=\"password\" value=\"\">"
"          <p class=\"text-danger\">Password is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"password\" class=\"col-sm-2 control-label\">Role</label>"
"        <div class=\"col-sm-10\">"
"          <label class=\"checkbox-inline\"><input type=\"radio\" value=\"ADMIN\" id=\"role1\" name=\"role\">Admin</label>"
"          <label class=\"checkbox-inline\"><input type=\"radio\" value=\"USER\" id=\"role2\" name=\"role\">User</label>"
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <input type=\"hidden\" id=\"id\" name=\"id\" value=\"1\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/user/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Update user\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
-
> POST /user/update/1
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /user/list
""
]


com.poseidon.app.http.controller.UserControllerTest.validate=[
-
> POST /user/validate
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Add New User</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/user/validate\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"fullname\" class=\"col-sm-2 control-label\">Full Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"fullname\" placeholder=\"Full Name\" class=\"col-4\" name=\"fullname\" value=\"\">"
"          <p class=\"text-danger\">Full Name is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"username\" class=\"col-sm-2 control-label\">User Name</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"username\" placeholder=\"User Name\" class=\"col-4\" name=\"username\" value=\"\">"
"          <p class=\"text-danger\">Username is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"password\" class=\"col-sm-2 control-label\">Password</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"password\" id=\"password\" placeholder=\"Password\" class=\"col-4\" name=\"password\" value=\"\">"
"          <p class=\"text-danger\">Password is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"password\" class=\"col-sm-2 control-label\">Role</label>"
"        <div class=\"col-sm-10\">"
"          <label class=\"checkbox-inline\"><input type=\"radio\" value=\"ADMIN\" id=\"role1\" name=\"role\">Admin</label>"
"          <label class=\"checkbox-inline\"><input type=\"radio\" value=\"USER\" id=\"role2\" name=\"role\">User</label>"
"          <p class=\"text-danger\">Role is mandatory</p>"
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/user/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Add user\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
-
> POST /user/validate
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /user/list
""
]